test/test.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"crypto/sha512"
+	"encoding/hex"
+	"fmt"
+	ledger "github.com/spacemeshos/go-ledger-sdk"
+	"github.com/spacemeshos/ed25519"
+)
+
+func uint64_to_buf(value uint64) []byte {
+	data := make([]byte, 8)
+	data[0] = byte((value >> 56) & 0xff)
+	data[1] = byte((value >> 48) & 0xff)
+	data[2] = byte((value >> 40) & 0xff)
+	data[3] = byte((value >> 32) & 0xff)
+	data[4] = byte((value >> 24) & 0xff)
+	data[5] = byte((value >> 16) & 0xff)
+	data[6] = byte((value >>  8) & 0xff)
+	data[7] = byte((value      ) & 0xff)
+	return data
+}
+
+func do_test(device *ledger.HidDevice) {
+	if err := device.Open(); err != nil {
+		return
+	}
+	defer device.Close()
+
+	version, err := device.GetVersion()
+	if err != nil {
+		fmt.Printf("get version ERROR: %v\n", err)
+	} else {
+		fmt.Printf("version: %+v\n", version)
+	}
+	publicKey, err := device.GetExtendedPublicKey(ledger.StringToPath("44'/540'/0'/0/0'"))
+	if err != nil {
+		fmt.Printf("get public key ERROR: %v\n", err)
+	} else {
+		fmt.Printf("public key: %+v\n", publicKey)
+	}
+	address, err := device.GetAddress(ledger.StringToPath("44'/540'/0'/0/0'"))
+	if err != nil {
+		fmt.Printf("get address ERROR: %v\n", err)
+	} else {
+		fmt.Printf("address: %+v\n", address)
+	}
+	err = device.ShowAddress(ledger.StringToPath("44'/540'/0'/0/1'"))
+	if err != nil {
+		fmt.Printf("show address ERROR: %v\n", err)
+	} else {
+		fmt.Printf("show address: OK\n")
+	}
+
+	tx := make([]byte, 0)
+	var bin []byte
+	bin, _ = hex.DecodeString("0000000000000000000000000000000000000000000000000000000000000000") // network id
+	tx = append(tx, bin...)
+	tx = append(tx, 0) // coin transaction with ed
+	tx = append(tx, uint64_to_buf(1)...) // nonce
+	bin, _ = hex.DecodeString("0000000000000000000000000000000000000000") // recepient
```

Comment:
would be nice to use some non-zero value so we can see that encoding works (this should be printed, so we can easily confirm when running the test.

(not just on this line, all the values should be more arbitrary and not zeros)

Reply:
Done!